igcollect/logfile_values.py

```diff
@@ -65,28 +65,66 @@ def get_timeshift(self):
                     return int(self.period[:-len(unit)].strip()) * mul
         return 0
 
+    def extract_value_from_field(self, field):
+        '''
+        Extract numeric value from field, handling key-value pairs and plain values
+        '''
```

Comment:
It is fine to use single or double quotes in a project or script as long as it is consistent but doc strings should always be triple quotes. 

> For consistency, always use """triple double quotes""" around docstrings. Use r"""raw triple double quotes""" if you use any backslashes in your docstrings.
https://peps.python.org/pep-0257/